sklearn/preprocessing/_data.py

```diff
@@ -3205,6 +3205,19 @@ def power_transform(X, method='yeo-johnson', *, standardize=True, copy=True):
 
     Read more in the :ref:`User Guide <preprocessing_transformer>`.
 
+    .. warning:: Risk of data leak
+
+        Do not use :func:`~sklearn.preprocessing.power_transform` unless you
+        know what you are doing. A common mistake is to apply it to the entire
+        data *before* splitting into training and test sets. This will bias the
+        model evaluation because information would have leaked from the test
+        set to the training set.
+        In general, we recommend using
+        :class:`~sklearn.preprocessing.PowerTransformer` within a
+        :ref:`Pipeline <pipeline>` in order to prevent most risks of data
+        leaking:
```

Comment:
add a "for example" maybe?

Reply:
Thanks, @adrinjalali! I am not sure what you mean by a "for example"?

Reply:
@abbiepopa I was able to correct what @adrinjalali meant by his help and the PR is already Merged

Reply:
Thanks @GeorMurk ! I still see the PR as open? cc @adrinjalali

Reply:
you're welcome @abbiepopa maybe you could close the PR or @adrinjalali may also close the PR.
Here is where the PR was solved and Merged #17535